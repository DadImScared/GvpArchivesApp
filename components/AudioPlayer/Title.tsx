import * as React from "react";

import { View } from "react-native";
import Ticker from "../../components/Marquee";

interface ITitleProps {
  getSongName: () => string;
}

export const Title: React.FunctionComponent<ITitleProps> = ({ getSongName }) => (
  <View style={{ justifyContent: "center", alignItems: "center" }}>
    <Ticker duration={5000} bounce={false} loop={true}>
      {getSongName()}
    </Ticker>
  </View>
);

export default Title;
