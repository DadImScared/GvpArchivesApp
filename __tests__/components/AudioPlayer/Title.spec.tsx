import * as React from "react";
import * as renderer from "react-test-renderer";

import Title from "../../../components/AudioPlayer/Title";

describe("Title", () => {
  it("should render", () => {
    const getSongName = jest.fn(() => "song name here");
    const tree = renderer.create(<Title getSongName={getSongName} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
