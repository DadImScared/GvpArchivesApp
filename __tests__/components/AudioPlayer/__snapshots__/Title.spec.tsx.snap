// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Marquee
    bounce={false}
    duration={5000}
    loop={true}
  >
    song name here
  </Marquee>
</View>
`;
